src/openmc2dolfinx/vtk_reader.py

```diff
@@ -1,23 +1,167 @@
+from mpi4py import MPI
+
+import basix
 import dolfinx
+import numpy as np
 import pyvista
+import pyvista.core.pointset
+import ufl
+from dolfinx.mesh import create_mesh
 
+__all__ = ["StructuredGridReader", "UnstructuredMeshReader"]
 
-class VTKReader:
-    """"""
 
-    def __init__(self):
-        pass
+class OpenMC2dolfinx:
+    """
+    Base OpenMC2Dolfinx Mesh Reader
+
+    Reads an OpenMC .vtk results file mesh and converts the data into a
+    dolfinx.fem.Function
 
+    Args:
+        filename: the filename
 
-class StructuredGridReader(VTKReader):
-    """"""
+    Attributes:
+        filename: the filename
+        grid: the mesh and results from the OpenMC .vtk file
+        connectivity: The OpenMC mesh cell connectivity
+        dolfinx_mesh: the dolfinx mesh
+    """
+
+    grid: pyvista.core.pointset.UnstructuredGrid | pyvista.core.pointset.StructuredGrid
+    connectivity: np.ndarray
+    dolfinx_mesh: dolfinx.mesh.Mesh
 
     def __init__(self):
-        pass
+        self.grid = None
+        self.cell_connectivity = None
+        self.dolfinx_mesh = None
```

Comment:
This isn't tested. That's because this actual class is never instantiated. We probably don't need an __init__ for this